include/alpaka/mem/buf/BufOacc.hpp

```diff
@@ -0,0 +1,652 @@
+        };
+        //#############################################################################
+        //! The BufOacc device get trait specialization.
+        template<
+            typename TElem,
+            typename TDim,
+            typename TIdx>
+        struct GetDev<
+            BufOacc<TElem, TDim, TIdx>>
+        {
+            ALPAKA_FN_HOST static auto getDev(
+                BufOacc<TElem, TDim, TIdx> const & buf)
+            -> DevOacc
+            {
+                return (*buf).m_dev;
+            }
+        };
+
+        //#############################################################################
+        //! The BufOacc dimension getter trait specialization.
+        template<
+            typename TElem,
+            typename TDim,
+            typename TIdx>
+        struct DimType<
+            BufOacc<TElem, TDim, TIdx>>
+        {
+            using type = TDim;
+        };
+
+        //#############################################################################
+        //! The BufOacc memory element type get trait specialization.
+        template<
+            typename TElem,
+            typename TDim,
+            typename TIdx>
+        struct ElemType<
+            BufOacc<TElem, TDim, TIdx>>
+        {
+            using type = TElem;
+        };
+    }
+
+    namespace extent
+    {
+        namespace traits
+        {
+            //#############################################################################
+            //! The BufOacc extent get trait specialization.
+            template<
+                typename TIdxIntegralConst,
+                typename TElem,
+                typename TDim,
+                typename TIdx>
+            struct GetExtent<
+                TIdxIntegralConst,
+                BufOacc<TElem, TDim, TIdx>,
+                typename std::enable_if<(TDim::value > TIdxIntegralConst::value)>::type>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto getExtent(
+                    BufOacc<TElem, TDim, TIdx> const & extent)
+                -> TIdx
+                {
+                    return extent.extentElements()[TIdxIntegralConst::value];
+                }
+            };
+        }
+    }
+
+    namespace traits
+    {
+        //#############################################################################
+        //! The BufOacc native pointer get trait specialization.
+        template<
+            typename TElem,
+            typename TDim,
+            typename TIdx>
+        struct GetPtrNative<
+            BufOacc<TElem, TDim, TIdx>>
+        {
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST static auto getPtrNative(
+                BufOacc<TElem, TDim, TIdx> const & buf)
+            -> TElem const *
+            {
+                return (*buf).m_pMem;
```

Comment:
Why not? `buf->m_pMem`? If you provide `operator*` and `operator->` should be there as well.

Reply:
@jkelling